cmd/api/src/database/auth.go

```diff
@@ -282,6 +282,15 @@ func (s *BloodhoundDB) UpdateUser(ctx context.Context, user model.User) error {
 			return err
 		}
 
+		// AuthSecret must be manually deleted and then set to nil again to prevent recreation
+		if user.AuthSecret == nil {
+			if err := tx.Unscoped().Model(&user).WithContext(ctx).Association("AuthSecret").Unscoped().Clear(); err != nil {
+				return err
+			}
+
+			user.AuthSecret = nil
+		}
+
```

Comment:
This feels a bit like we are swimming up stream. I'm hopeful we can find a smoother way if we pair!

Reply:
After pairing, we opted for the more verbose but single pathway of calling the DeleteAuthSecret method to ensure audit logs are maintained